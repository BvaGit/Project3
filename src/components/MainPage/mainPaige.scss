@import '../../utils/variables.scss';

.header{
    background: $button-color;
    color: $button-text;
    padding: $padding_lg;
    @include flex(row, center,_);
    &__appName{
        margin-right: $margin_md;
        margin-top: auto;
        margin-bottom: auto;
    }
    &__logo{
        width: 60px;
        text-align: center;
    }
}
.rooms__box{
    width: 30%;
    min-height: 90%;
    position: absolute;
    border: $border solid 1px;
}