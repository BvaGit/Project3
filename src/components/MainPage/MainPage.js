import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";

import RoomsHeader from "../RoomsHeader";
import Rooms from "../Rooms";
import AddRoom from "../AddRoom";
import MessagesContainer from "../MessagesContainer";
import { addToken, myAccountGET } from "../../store/user/asyncActions";
import { getActiveRoom } from "../../store/rooms/selectors";

import ConnectSocket from "../Socket";

import "./mainPaige.scss";

const MainPage = () => {
  const dispatch = useDispatch();

  const chatId = useSelector(getActiveRoom);

    useEffect(() => {
        dispatch(addToken());
        dispatch(myAccountGET());
    }, []);

  return (
    <div className="inner">
      <header className="header">
        <img
          className="header__logo"
          src="../../../public/assets/images/logo.png"
        />
        <h1 className="header__appName">Blabber</h1>
      </header>
      <div className="rooms__box">
        <RoomsHeader />
        <Rooms />
        <AddRoom />
        <ConnectSocket />
      </div>
      <MessagesContainer chatId={chatId} />
    </div>
  );
};

export default MainPage;
