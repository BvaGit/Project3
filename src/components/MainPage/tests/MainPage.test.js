import React from 'react';
import { shallow, mount } from 'enzyme';
import MainPage from '../MainPage';
import { connectRouter } from "connected-react-router";
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { IntlProvider } from "react-intl";



const mockStore = configureStore();
const store = mockStore({
    auth : {
        fields: {
          login: "",
          password: "",
        }, 
        errors: {
          login: "",
          password: "",
        }
      },
    rooms: {
        sendingMessage: false,
        messages: {
          0: [
            {
              text: "Welcome!",
            },
          ],
        },
        socket: null,
        rooms: {},
        activeRoom: "",
      },
      user:{
        user: { id: null, token: null },
      },
      users: {
        users: [],
      },
      registration: {
        fields:{
          login: '',
          password: '',
          confirm_password:""
        },
        errors:{
          login:"",
          password:"",
          confirm_password:""
        }
      },
        router: connectRouter(history),
        language: {
            value: 'en'
        },
    });


describe('MainPage', () => {
    it("Should match snapshot", () => {
        const component = shallowBiggerSmart(<MainPage />, store)
        expect(component.html()).toMatchSnapshot()
    })
    // it("Soulr render mainPage", ()=>{
    //   const component = mountBiggerSmart(<MainPage />, store)
    //   console.log(component.debug())
    // })
})


export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider >core</IntlProvider>);
};

export const mountBiggerSmart = (component, store) => {
  const core = store
      ? <Provider store={store}>{component}</Provider>
      : component;
  return mount(<IntlProvider locale="en" messages={{}}>{core}</IntlProvider>);
};