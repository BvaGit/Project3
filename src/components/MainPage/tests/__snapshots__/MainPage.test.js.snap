// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Should match snapshot 1`] = `"<div class=\\"inner\\"><header class=\\"header\\"><img class=\\"header__logo\\" src=\\"/assets/images/logo.png\\"/><h1 class=\\"header__appName\\">Blabber</h1></header><div class=\\"rooms__box\\"><div class=\\"rooms-header\\"><div class=\\"rooms-header__usermenu\\"><div class=\\"rooms-header__userinfo\\"><img class=\\"rooms-header__avatar\\" src=\\"/assets/images/1-36-1024x1024.jpg\\"/><div class=\\"rooms-header__userName\\"></div></div><div class=\\"rooms-header__settingsmenu\\"><div><img class=\\"rooms-header__settings\\" src=\\"/assets/images/seting_2.png\\"/></div><img class=\\"rooms-header__logout\\" src=\\"/assets/images/logout_2.png\\"/></div></div><div class=\\"rooms-header__search\\"><label class=\\"inputs__label \\">search<input title=\\"Enter room name\\" placeholder=\\"searchPlaceholder\\" class=\\"inputs__input rooms-header__input\\"/></label></div></div><div class=\\"rooms\\"><div class=\\"rooms__message\\">No chats</div></div><div><div class=\\"add-room\\">+</div></div></div><div class=\\"message-box message-box--empty\\">Welcome, <p class=\\"message-box--empty__user\\"></p>!</div></div>"`;
